compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -725,41 +725,24 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
 
         self.super_projection_elem(place_local, proj_base, elem, context, location);
 
-        match elem {
```

Comment:
Why did you convert this `match` to an `if`? It was a `match` deliberately so that when a new `ProjectionElem` is added, one needs to explicitly consider what to do here.

Reply:
Every other branch would be eliminated. I assumed an `if` was appropriate given only one item would be checked.

Reply:
Please keep it a `match`, so that we still have exhaustiveness checking.

Reply:
Will do. I'll just have the existing branches do nothing in that case.

Reply:
Yes, exactly. :)